plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Payments.php

```diff
@@ -219,6 +224,33 @@ private function is_woopayments_onboarded(): bool {
 		return false;
 	}
 
+	/**
+	 * Check if WooPayments has a live account onboarding in progress.
+	 *
+	 * @return bool
+	 */
+	private function has_woopayments_live_account_in_progress() {
+		if ( $this->is_woopayments_onboarded() ) {
+			return false;
+		}
+
+		$woopayments_provider = $this->get_woopayments_provider();
+		// We should have the WooPayments provider, but if not, return false.
+		if ( ! $woopayments_provider ) {
+			return false;
+		}
+
+		// Check the provider's state to determine if a live account onboarding is in progress.
+		if ( ! empty( $woopayments_provider['onboarding']['state']['started'] ) &&
+			empty( $woopayments_provider['onboarding']['state']['completed'] ) &&
```

Comment:
`completed` is already checked at the beginning when checking if it is onboarded. Just checking that the onboarding started should be enough (of course, after all the other conditions didn't cause a bail).

Reply:
It makes sense to check only if the onboarding started as the previous conditions are covering the other parts. Updated!